src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final int NAME = 0;
+    private static final int NUMBER = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> fileContent = readFile(fromFileName);
+        String report = getResult(fileContent);
+        writeToFile(toFileName, report);
+    }
+
+    private static List<String> readFile(String fromFileName) {
+        try {
+            List<String> lines = Files.readAllLines(Path.of(fromFileName));
+            return lines;
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data");
+        }
+    }
+
+    private static String getResult(List<String> lines) {
+        int supplySum = 0;
+        int buySum = 0;
+        for (String line : lines) {
+            String [] parts = line.split(",");
+            if (parts[NAME].equals("supply")) {
+                supplySum += Integer.parseInt(parts[NUMBER]);
+            }
+            if (parts[NAME].equals("buy")) {
+                buySum += Integer.parseInt(parts[NUMBER]);
+            }
```

Comment:
If you have strange strings or numbers in the code, it’s better to declare them as constants. The name of the constant should display this object’s purpose. E. G. COMA, BUY, SUPPLY, etc.